### PR TITLE
Log errors in form submission attempts

### PR DESCRIPTION
**Pull request by Thrillberg**
## Summary
This PR does a couple things:
1. In the `BenefitsIntakeStatusJob`, only collect form submissions that _only_ have pending statuses.
2. Record error messages on the form submission attempt.

## Related issue(s)
https://app.zenhub.com/workspaces/vagov-product-team-forms-634853151f5c6000165942bc/issues/gh/department-of-veterans-affairs/va.gov-team-forms/1613

### CODE REVIEW

File: app/sidekiq/benefits_intake_status_job.rb [hunk 1]
```diff
@@ -21,10 +21,16 @@ def initialize(batch_size: BATCH_SIZE)
...
+                                .joins(:form_submission_attempts)
+                                .where(form_submission_attempts: { aasm_state: %w[vbms failure] })
     pending_form_submissions = FormSubmission
                                .joins(:form_submission_attempts)
                                .where(form_submission_attempts: { aasm_state: 'pending' })
-    total_handled, result = batch_process(pending_form_submissions)
+    # We're calculating the resolved_form_submissions and subtracting them because it is possible for a FormSubmission
+    # to have two (or more) attempts, one 'pending' and the other 'vbms'. In such cases we don't want to include
+    # that FormSubmission because it has been resolved.
+    total_handled, result = batch_process(pending_form_submissions - resolved_form_submissions)
```
Comment (Thrillberg) [inline app/sidekiq/benefits_intake_status_job.rb:33]: If any ActiveRecord wizards can improve/condense this query, I'm all ears!!

Comment (freeheeling) [reply app/sidekiq/benefits_intake_status_job.rb:32]: Avoid loading both sets of objects into memory. Filter the pending submissions directly in the database. For example:
```
form_submissions_and_attempts = FormSubmission.joins(:form_submission_attempts)

resolved_form_submission_ids = form_submissions_and_attempts
  .where(form_submission_attempts: { aasm_state: %w[vbms failure] })
  .pluck(:id)

pending_form_submissions = form_submissions_and_attempts
  .where(form_submission_attempts: { aasm_state: 'pending' })
  .where.not(id: resolved_form_submission_ids)

total_handled, result = batch_process(pending_form_submissions)
```

Comment (Thrillberg) [reply app/sidekiq/benefits_intake_status_job.rb]: Thanks @freeheeling ! Updated the branch.

File: app/sidekiq/benefits_intake_status_job.rb [hunk 2]
```diff
@@ -88,9 +104,9 @@ def handle_response(response, pending_form_submissions)
   end
   # rubocop:enable Metrics/MethodLength
 
-  def log_result(result, form_id, uuid, time_to_transition = nil)
+  def log_result(result, form_id, uuid, time_to_transition = nil, error_message = nil)
     StatsD.increment("#{STATS_KEY}.#{form_id}.#{result}")
     StatsD.increment("#{STATS_KEY}.all_forms.#{result}")
-    Rails.logger.info('BenefitsIntakeStatusJob', result:, form_id:, uuid:, time_to_transition:)
+    Rails.logger.info('BenefitsIntakeStatusJob', result:, form_id:, uuid:, time_to_transition:, error_message:)
```
Comment (pennja) [inline app/sidekiq/benefits_intake_status_job.rb:110]: Do we want both successes and failures to both be `Rails.logger.info` or do we want failures to result in an error log?

Comment (Thrillberg) [reply app/sidekiq/benefits_intake_status_job.rb:110]: Oh good call! I'll change that now.

**Review by pennja — APPROVED** [review]
Looks good, great work! 👍

**Pull request merged**